clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```diff
@@ -16,93 +16,103 @@
 import org.apache.kafka.common.utils.Bytes;
 
 import java.nio.ByteBuffer;
+import java.util.Map;
 
 /**
  * Factory for creating serializers / deserializers.
  */
 public class Serdes {
 
-    static public final class LongSerde implements Serde<Long> {
-        @Override
-        public Serializer<Long> serializer() {
-            return new LongSerializer();
-        }
+    static private class WrapperSerde<T> implements Serde<T> {
+        protected Serializer<T> serializer;
+        protected Deserializer<T> deserializer;
```

Comment:
Hmm, why don't we have this as private final and pass them as constructor parameters from the subclasses?